src/org/oidc/service/AbstractService.java

```diff
@@ -151,11 +178,64 @@ public abstract void updateServiceContext(
      * @param stateKey the key that identifies the State object
      * @return the parsed and to some extent verified response
      **/
-    public Message parseResponse(
-            String responseBody,
-            SerializationType serializationType,
-            String stateKey) {
-        return null;
+    public Message parseResponse(String responseBody, SerializationType serializationType, String stateKey) throws Exception {
+        if(serializationType == null) {
+            serializationType = this.serializationType;
+        }
+
+        if(SerializationType.URL_ENCODED.equals(serializationType)) {
+            responseBody = ServiceUtil.getUrlQueryReference(responseBody);
+        }
+
+        Message response = null;
+        try {
+            response = this.responseMessage.deserialize(responseBody, serializationType);
+        } catch (Exception e) {
```

Comment:
Ideally you shouldn't catch Exception. You should catch individual exception that you need to catch. Refer http://go/java-practices/exceptions#catch_exception